frontend/src/components/SearchForm.js

```diff
@@ -1,42 +1,77 @@
-import { Fragment } from "react";
+import { Fragment, useState } from "react";
 import '../sass/components/searchform.scss';
 import '../sass/base/typography.scss';
 import icon from '../assets/search-icon.png';
 
-const SearchForm = () => (
+const SearchForm = () => {
+    const [radioDifficulty, setRadioDifficulty] = useState('beginner');
+
+    const onSetRadioDifficulty = (value) => {
+        setRadioDifficulty(value);
+    }
+
+    const difficultyHandler = (value) => {
+        onSetRadioDifficulty(value);
+    }
+
```

Comment:
`onSetRadioDifficulty` and `difficultyHandler` are duplicates. They are both doing the same thing. One will have to go. Since this component is managing its own state, there is no need for the subscriber-publisher function patter. So, just one of these functions is fine here. Meaning, lose one and keep the other. Whichever one you delete remember to update the code where you previously used it.

Reply:
Fixed!